core/cas-server-core-services/src/main/java/org/apereo/cas/services/DomainServicesManager.java

```diff
@@ -0,0 +1,197 @@
+package org.apereo.cas.services;
+
+import org.apereo.cas.authentication.principal.Service;
+import org.apereo.cas.support.events.service.CasRegisteredServiceDeletedEvent;
+import org.apereo.cas.support.events.service.CasRegisteredServiceSavedEvent;
+import org.apereo.inspektr.audit.annotation.Audit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationEvent;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.scheduling.annotation.Scheduled;
+
+import javax.annotation.PostConstruct;
+import java.io.Serializable;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Predicate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * Default implementation of the {@link ServicesManager} interface.
```

Comment:
Need better comments here please, essentially what you have put down as the description for the PR :)

Also author and version tags need adjusting.